sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/Sgv2RequestHandler.java

```diff
@@ -1,167 +0,0 @@
-            sc, Response.Status.UNAUTHORIZED, "Unauthenticated gRPC operation", grpcE.getMessage());
-      case UNAVAILABLE:
-        // Should this be SERVICE_UNAVAILABLE (503) or BAD_GATEWAY (502)?
-        return grpcResponse(
-            sc,
-            Response.Status.SERVICE_UNAVAILABLE,
-            "gRPC service unavailable",
-            grpcE.getMessage());
-      case UNIMPLEMENTED:
-        return grpcResponse(
-            sc,
-            Response.Status.NOT_IMPLEMENTED,
-            "Unimplemented gRPC operation",
-            grpcE.getMessage());
-
-        // And then codes we ... don't know how to handle
-      case OK: // huh?
-
-      case CANCELLED:
-      case UNKNOWN:
-      case DEADLINE_EXCEEDED:
-      case ALREADY_EXISTS:
-      case RESOURCE_EXHAUSTED:
-      case ABORTED:
-      case OUT_OF_RANGE:
-      case INTERNAL:
-      case DATA_LOSS:
-        break;
-    }
-    return grpcResponse(
-        sc, Response.Status.INTERNAL_SERVER_ERROR, "Unhandled gRPC failure", grpcE.getMessage());
-  }
-
-  private static Response failResponseForOtherException(Exception e) {
-    if (e instanceof IllegalArgumentException) {
-      logger.info("Bad request (IllegalArgumentException->BAD_REQUEST): {}", e.getMessage());
-      return Response.status(Response.Status.BAD_REQUEST)
-          .entity(
-              new RestServiceError(
-                  "Bad request: " + e.getMessage(), Response.Status.BAD_REQUEST.getStatusCode()))
-          .build();
-    }
-    if (e instanceof JsonProcessingException) {
-      logger.info(
-          "Invalid JSON payload (JsonProcessingException->BAD_REQUEST): {}", e.getMessage());
-      return Response.status(Response.Status.BAD_REQUEST)
-          .entity(
-              new RestServiceError(
-                  "Role unauthorized for operation: " + e.getMessage(),
-                  Response.Status.BAD_REQUEST.getStatusCode()))
-          .build();
-    }
-    if (e instanceof ExecutionException) {
-      //      if (ee.getCause() instanceof
-      // org.apache.cassandra.stargate.exceptions.UnauthorizedException) { }
-
-      // Do log underlying Exception with Stack trace since this is unknown, unexpected:
-      logger.error("Unrecognized error when executing request", e);
-      return Response.status(Response.Status.INTERNAL_SERVER_ERROR)
-          .entity(
-              new RestServiceError(
-                  "Server error: " + e.getMessage(),
-                  Response.Status.INTERNAL_SERVER_ERROR.getStatusCode()))
-          .build();
-    }
-    // Do log underlying Exception with Stack trace since this is unknown, unexpected:
-    logger.error("Unrecognized error when executing request", e);
-    return Response.status(Response.Status.INTERNAL_SERVER_ERROR)
-        .entity(
-            new RestServiceError(
-                "Server error: " + e.getMessage(),
-                Response.Status.INTERNAL_SERVER_ERROR.getStatusCode()))
-        .build();
```

Comment:
Dropwizard already does this by default, see `DefaultExceptionMapper.logException`.

I don't think we should special case `IllegalArgumentException` et al, those should be caught by the application code and handled when they indicate a business error. Otherwise we might encounter unexpected ones and accidentally leak sensitive info in an error message.